src/components/header/OverflowEdgeIndicator.js

```diff
@@ -2,20 +2,34 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
 import React, { PureComponent, PropTypes } from 'react';
 import classNames from 'classnames';
 
 import './OverflowEdgeIndicator.css';
 
+type Props = {
+  className: string,
+  children: number,
+};
+
 class OverflowEdgeIndicator extends PureComponent {
-  constructor(props) {
+  props: Props;
+  state = {
+    overflowsOnTop: false,
+    overflowsOnRight: false,
+    overflowsOnBottom: false,
+    overflowsOnLeft: false,
+  };
```

Comment:
I adapted it from the example given in [Flow + React](https://flow.org/en/docs/frameworks/react/), but it didn't compile when I tried to run it. My bad! I'll fix it.